src/matrixbases.jl

```diff
@@ -55,6 +56,72 @@ function represent(basis::Type{T}, m::Matrix{<:Number}) where T<:AbstractMatrixB
     represent(basis{typeof(m)}(d), m)
 end
 
-function combine(basis::T, v::Vector{<:Number}) where T<:AbstractMatrixBasis
+function combine(basis::AbstractMatrixBasis{T}, v::Vector{<:Number}) where T<:AbstractMatrix{<:Number}
     sum(basis.iterator .* v)
+end
+
+"""
+$(SIGNATURES)
+
+"""
+abstract type AbstractMatrixBasisIterator{T<:AbstractMatrix} end
+struct ChannelBasisIterator{T} <: AbstractMatrixBasisIterator{T} 
+    idim::Int 
+    odim::Int
+    hitr::HermitianBasisIterator{T}
+    function ChannelBasisIterator{T}(idim::Int, odim::Int) where T<:AbstractMatrix{<:Number}
+        new(idim, odim, HermitianBasisIterator{T}(idim))
+    end
+end
+
+abstract type AbstractChannelBasis{T} <: AbstractMatrixBasis{T} end 
+struct ChannelBasis{T} <: AbstractMatrixBasis{T} 
+    iterator::ChannelBasisIterator{T}
+    function ChannelBasis{T}(idim::Integer, odim::Integer) where T<:AbstractMatrix{<:Number}
+        new(ChannelBasisIterator{T}(idim, odim)) 
+    end
+end
+channelbasis(T::Type{<:AbstractMatrix{<:Number}}, idim::Int, odim::Int=idim) = ChannelBasis{T}(idim, odim)
+
+channelbasis(idim::Int, odim::Int=idim) = channelbasis(Matrix{ComplexF64}, idim, odim)
+
+function iterate(itr::ChannelBasisIterator{T}, state=(1,1,1,1)) where T<:AbstractMatrix{<:Number}
+    idim = itr.idim
+    odim = itr.odim
+    hitr = itr.hitr
+    (a, c, b, d) = state
+    (a == odim && c == odim && d == 2) && return nothing
+
+    Tn = eltype(T)
+    if a > c
+        x = (ketbra(T, a, c, odim) ⊗ ketbra(T, b, d, idim) + ketbra(T, c, a, odim) ⊗ ketbra(T, d, b, idim)) / sqrt(Tn(2))
```

Comment:
This is almost the same as line 99. Maybe we can write them as one expression. The same holds for the cases in the HermiatianBasis